### ISSUE TITLE
CTD when player is about to go down in battle

### ISSUE THREAD

**Issue by madrooky** [Norbivar/RTSCamera#19]
Like the title says, this issue is bothering me for quite some time. I can't say that RTS is for sure the cause but that is my hunch.
I don't get any error messages either since when it happens the game just closes.
And this is why i think RTS is the cause, it crashes in situations where i see a final blow coming in battle, then the screen freezes for a second and just closes. I think RTS fails to switch to a companion after player death. But i can't prove it, i don't get any error messages. But that is the only thing that makes logical sense why this happens.
If someone can tell me how i can get an error report out of this, or make it somehow reproducable... I am out of ideas in this case.

**Comment by madrooky** [comment]
A few more Info, i am playing on 1.1.5, and use 4.1.23. 
I played a while without this mod and i did not have any crashes when going down in battle. Just activated it again, and at the end of a long siege, i was overwhelmed by the remaining defenders and guess what, CTD.

**Comment by madrooky** [comment]
I think i found the culprit.
It is the setting to prefer units of the same formation. When i unchecked this i can die without CTD.
Worth noting that i have set my char in group 8 which is usually not used for other units. 
Now it is easy to assume that this setting tried to switch to a unit in the formation of the player but did not account for the case that there is none, so the game crashes...

**Comment by madrooky** [comment]
Ok in siege the game still crashes when the char goes down.

**Comment by madrooky** [comment]
And i saw the other report about CTD when manually changing the char, i just tested it and can confirm it does cause basically the same CTD.

**Comment by Norbivar** [comment]
A stacktrace could be useful.

**Comment by madrooky** [comment]
> A stacktrace could be useful.

But how when the game just closes without any error? I would have provided the usual logs if they would show up.
I checked the vanilla logs too and with such a crash the log just ends...
If you can't reproduce by the description I don't know what to do. For me it was 100% reliable to reproduce.

**Comment by Norbivar** [comment]
Alright, let's give this another go.
Is it still happening? What versions are you on (bannerlord AND rtscam)? 
Do you use any other mods? It maybe could be a conflict, I dunno. 

If by any chance you could update to bannerlord 1.1.6 and use the newest release for that version (beware that I usually have two releases for actual mod version, since one goes to beta branch and one is the stable one) and see if it still happens, that would be lovely.

**Comment by madrooky** [comment]
> Alright, let's give this another go. Is it still happening? What versions are you on (bannerlord AND rtscam)? Do you use any other mods? It maybe could be a conflict, I dunno.
> 
> If by any chance you could update to bannerlord 1.1.6 and use the newest release for that version (beware that I usually have two releases for actual mod version, since one goes to beta branch and one is the stable one) and see if it still happens, that would be lovely.

Thanks for giving this attention, but I am currently not playing Bannerlord. I gave it another pause because of too many issues with mods in general and the lack of progress in the game. I have constructed a screenshot with all the mods and versions I used at the time of reporting. Luckily it is still installed. It does not include the inactive items I have disabled because of mentioned issues... it's a much longer list, unfortunately.

I would not suggest you should try all the mods. 
But at least the mod should in any case work in combination with the first 4 libraries, harmony, butterlib, UIX and MCM as they are a must have. Everything else is optional.

If you can't reproduce the issue in the current versions you might as well close the case.

When I find time and patience to give Bannerlord another try, and go through the dreadful process and update everything, I will be glad if I can help you out actively if the issue persists.

![image](https://github.com/Norbivar/RTSCamera/assets/131664307/76785f52-b2b6-4aab-91f1-1227a4759e43)

**Comment by Norbivar** [comment]
By any chance, do you have the automatic switch on death enabled?
I did encounter a stack overflow with that, so I will handle that these issues are the same.

**Comment by madrooky** [comment]
> Alright, let's give this another go. Is it still happening? What versions are you on (bannerlord AND rtscam)? Do you use any other mods? It maybe could be a conflict, I dunno.
> 
> If by any chance you could update to bannerlord 1.1.6 and use the newest release for that version (beware that I usually have two releases for actual mod version, since one goes to beta branch and one is the stable one) and see if it still happens, that would be lovely.

Thanks for giving this attention, but I am currently not playing Bannerlord. I gave it another pause because of too many issues with mods in general and the lack of progress in the game. I have constructed a screenshot with all the mods and versions I used at the time of reporting. Luckily it is still installed. It does not include the inactive items I have disabled because of mentioned issues... it's a much longer list, unfortunately.

I would not suggest you should try all the mods. 
But at least the mod should in any case work in combination with the first 4 libraries, harmony, butterlib, UIX and MCM as they are a must have. Everything else is optional.

If you can't reproduce the issue in the current versions you might as well close the case.

When I find time and patience to give Bannerlord another try, and go through the dreadful process and update everything, I will be glad if I can help you out actively if the issue persists.

![image](https://github.com/Norbivar/RTSCamera/assets/131664307/76785f52-b2b6-4aab-91f1-1227a4759e43)

**Comment by madrooky** [comment]
> the automatic



> By any chance, do you have the automatic switch on death enabled? I did encounter a stack overflow with that, so I will handle that these issues are the same.

Not sure if I understand you correctly, do you mean permanent death or something? I meant battle KO, i have turned off battle death for player and clan.

**Comment by Norbivar** [comment]
No, I meant that there is an option in the mod to automatically take control of the next troop as soon as you die (this means you don't have to press E).
